xdsl/dialects/memref.py

```diff
@@ -379,6 +377,49 @@ class Subview(Operation):
 
     irdl_options = [AttrSizedOperandSegments()]
 
+    @staticmethod
+    def from_static_parameters(
+        source: SSAValue | Operation,
+        source_element_type: Attribute,
+        source_shape: Sequence[int],
+        offsets: Sequence[int],
+        sizes: Sequence[int],
+        strides: Sequence[int],
+    ) -> Subview:
+
+        source = SSAValue.get(source)
+
+        layout_strides = [1]
+        for input_size in reversed(source_shape[1:]):
+            layout_strides.insert(0, layout_strides[0] * input_size)
+
+        layout_offset = sum([
+            stride * offset
+            for (stride, offset) in zip(layout_strides, offsets)
```

Comment:
```suggestion
            for stride, offset in zip(layout_strides, offsets)
```

Reply:
I feel pairs read better as it is (w parentheses)

Reply:
Happy with either

Reply:
I think we go without parenthesis everywhere else in the project, so I'd go without them